bin/collator/src/rpc.rs

```diff
@@ -368,7 +362,9 @@ where
             10,
             None,
             pending_create_inherent_data_providers,
-            Some(pending_consenus_data_provider),
+            Some(Box::new(fc_rpc::pending::AuraConsensusDataProvider::new(
```

Comment:
What about an alternative solution:

The trait is defined like:
```Rust
/// Consensus data provider, pending api uses this trait object for authoring blocks valid for any runtime.
pub trait ConsensusDataProvider<B: BlockT>: Send + Sync {
	/// Attempt to create a consensus digest.
	fn create_digest(
		&self,
		parent: &B::Header,
		data: &InherentData,
	) -> Result<Digest, sp_inherents::Error>;
}
```

We can make a type similar to how `WaitForAuraConsensus` is implemented.
Return `Err(...)` or dummy value if Aura runtime API isn't available, with an appropriate warn/error trace, and start returning data provided by the `AuraConsensusDataProvider` once Aura runtime API is available.

That way users aren't impacted.

Reply:
Yes, I thought about that. But don't we need this patch to be released first? If not I can do that

Reply:
We can proceed with "proper" solution immediately, no need to rush the fix today.

If someone wants to sync Shiden from scratch today, they can just use an older client.